src/language-js/print/ternary.js

```diff
@@ -0,0 +1,383 @@
+    ? "alternate"
+    : "falseType";
+  const testNodePropertyNames = isConditionalExpression
+    ? ["test"]
+    : ["checkType", "extendsType"];
+  const consequentNode = node[consequentNodePropertyName];
+  const alternateNode = node[alternateNodePropertyName];
+  const parts = [];
+
+  // [prettierx] spaceInParens option support (...)
+  const parenSpace = options.spaceInParens ? " " : "";
+  const parenLine = options.spaceInParens ? line : softline;
+
+  // We print a ConditionalExpression in either "JSX mode" or "normal mode".
+  // See tests/jsx/conditional-expression.js for more info.
+  let jsxMode = false;
+  const parent = path.getParentNode();
+  const isParentTest =
+    parent.type === node.type &&
+    testNodePropertyNames.some((prop) => parent[prop] === node);
+  let forceNoIndent = parent.type === node.type && !isParentTest;
+
+  // Find the outermost non-ConditionalExpression parent, and the outermost
+  // ConditionalExpression parent. We'll use these to determine if we should
+  // print in JSX mode.
+  let currentParent;
+  let previousParent;
+  let i = 0;
+  do {
+    previousParent = currentParent || node;
+    currentParent = path.getParentNode(i);
+    i++;
+  } while (
+    currentParent &&
+    currentParent.type === node.type &&
+    testNodePropertyNames.every(
+      (prop) => currentParent[prop] !== previousParent
+    )
+  );
+  const firstNonConditionalParent = currentParent || parent;
+  const lastConditionalParent = previousParent;
+
+  if (
+    isConditionalExpression &&
+    (isJsxNode(node[testNodePropertyNames[0]]) ||
+      isJsxNode(consequentNode) ||
+      isJsxNode(alternateNode) ||
+      conditionalExpressionChainContainsJsx(lastConditionalParent))
+  ) {
+    jsxMode = true;
+    forceNoIndent = true;
+
+    // Even though they don't need parens, we wrap (almost) everything in
+    // parens when using ?: within JSX, because the parens are analogous to
+    // curly braces in an if statement.
+    const wrap = (doc) => [
+      ifBreak("("),
+      indent([softline, doc]),
+      softline,
+      ifBreak(")"),
+    ];
+
+    // The only things we don't wrap are:
+    // * Nested conditional expressions in alternates
+    // * null
+    // * undefined
+    const isNil = (node) =>
+      node.type === "NullLiteral" ||
+      (node.type === "Literal" && node.value === null) ||
+      (node.type === "Identifier" && node.name === "undefined");
+
+    parts.push(
+      " ? ",
+      isNil(consequentNode)
+        ? print(consequentNodePropertyName)
+        : wrap(print(consequentNodePropertyName)),
+      " : ",
+      alternateNode.type === node.type || isNil(alternateNode)
+        ? print(alternateNodePropertyName)
+        : wrap(print(alternateNodePropertyName))
+    );
+  } else {
+    // normal mode
+    const part = [
+      line,
+      "? ",
+      // [prettierx] spaceInParens option support (...)
+      ...(consequentNode.type === node.type
+        ? [ifBreak("", "("), parenSpace]
```

Comment:
minor fix is needed here for consistency with 0.18.x:

```suggestion
        ? [ifBreak("", ["(", parenSpace])]
```

I will make this fix.